gems/pending/VixDiskLib/VixDiskLibServer.rb

```diff
@@ -113,15 +107,19 @@ def wait_for_status(status, secs_to_wait)
   #
   # Now write the URI used back to the parent (client) process to let it know which port was selected.
   #
-  writer = vddk.writer_to_caller
-  writer.puts "URI:#{uri_used}"
-  writer.flush
+  uri_writer = IO.open(3, 'w')
+  uri_writer.write "#{uri_used}"
+  uri_writer.flush
+  uri_writer.close
+
+  proc_reader = IO.open(4, File::RDONLY | File::SYNC)
   #
   # Trap Handlers useful for testing and debugging.
   #
-  trap('INT') { vddk.shut_down_service("Interrupt Signal received") && exit }
-  trap('TERM') { vddk.shut_down_service("Termination Signal received") && exit }
+  trap('INT') { vddk.shut_down_service("Interrupt Signal received"); exit }
+  trap('TERM') { vddk.shut_down_service("Termination Signal received"); exit }
 
+  Thread.new { $vim_log.info "Monitoring Thread"; proc_reader.read; $vim_log.info "Shutting down VixDiskLibServer - Worker has exited"; exit }
```

Comment:
hah.. yeah, I'm often that guy :smile:

I think I feel differently here just because the method's called `read`, and not `wait_for_eof` or something -- so there are two levels of indirection between what we're literally doing, and *why* we're doing it